internal/osbuildconfig/osbuildconfig_handler.go

```diff
@@ -0,0 +1,79 @@
+package osbuildconfig
+
+import (
+	"github.com/project-flotta/osbuild-operator/internal/manifests"
+	repositoryosbuild "github.com/project-flotta/osbuild-operator/internal/repository/osbuild"
+	repositoryosbuildconfig "github.com/project-flotta/osbuild-operator/internal/repository/osbuildconfig"
+	"github.com/project-flotta/osbuild-operator/internal/repository/secret"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type OSBuildConfigHandler struct {
+	OSBuildConfigRepository repositoryosbuildconfig.Repository
+	SecretRepository        secret.Repository
+	OSBuildRepository       repositoryosbuild.Repository
+	Scheme                  *runtime.Scheme
+	OSBuildCRCreator        manifests.OSBuildCRCreator
+}
+
+func NewOSBuildConfigHandler(osBuildConfigRepository repositoryosbuildconfig.Repository, osBuildRepository repositoryosbuild.Repository, secretRepository secret.Repository, scheme *runtime.Scheme, osBuildCRCreator manifests.OSBuildCRCreator) *OSBuildConfigHandler {
+	return &OSBuildConfigHandler{
+		OSBuildConfigRepository: osBuildConfigRepository,
+		OSBuildRepository:       osBuildRepository,
+		SecretRepository:        secretRepository,
+		Scheme:                  scheme,
+		OSBuildCRCreator:        osBuildCRCreator,
+	}
+}
+
+func (o *OSBuildConfigHandler) OSBuildConfigWebhookTriggers(w http.ResponseWriter, r *http.Request, namespace string, name string, secret string) {
+	logger := log.FromContext(r.Context())
+	logger.Info("OSBuildConfigWebhookTriggerHandler new request")
+
+	// TODO - understand how to get params and use the responses
+	osBuildConfig, err := o.OSBuildConfigRepository.Read(r.Context(), name, namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		logger.Error(err, "can't read OSBuildConfigCR")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if osBuildConfig.Spec.Triggers.WebHook == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	secretName := osBuildConfig.Spec.Triggers.WebHook.SecretReference.Name
+	secretVal := secret
+	webhookSecret, err := o.SecretRepository.Read(r.Context(), secretName, namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		logger.Error(err, "can't read secret")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if string(webhookSecret.Data["WebHookSecretKey"]) != secretVal {
```

Comment:
according to the API, the secret provided to this method is the name of the secret. it isn't the value as it is being referred to here.
is the expectation to have the content of a secret part of the path?
also, if there are assumptions about the secret, e.g. the requirement to have a key named `WebHookSecretKey` it should be well documented.

Reply:
I changed it, it was a bug that I fix but forgot to update the API 
It's a secret value - please read the new description, is it more clear now?

Reply:
No, the secret provided is the value of the key `WebHookSecretKey`. It is documented here: https://github.com/project-flotta/osbuild-operator/blob/main/config/crd/bases/osbuilder.project-flotta.io_osbuildconfigs.yaml#L179

Reply:
I checked back at GitHub and found that it does not allow adding custom fields into the payload of a webhook call. As a result, any information required by the receiver must be put into the URL. 
GitHub will send it's own payload as either JSON or URL Encoded.

Reply:
I changed it, now it's part of the header (openapi v3 has 4 types of parameters- path, query, header and cookie, more details can be found here  
https://swagger.io/docs/specification/describing-parameters/#header-parameters )
Please re-review @masayag @eloycoto @jakub-dzon